src/components/renderer.tsx

```diff
@@ -36,7 +36,14 @@ export function useWidgetRendererWithRef<Configuration>(
   }
 
   const scheme = props.uiMessageWebviewUrlScheme || "mx"
-  const handler = makeRequestInterceptor(url, scheme, props, dispatchEvent)
+  const handler = makeRequestInterceptor(url, scheme, {
+    onIntercept: (url) => {
+      dispatchEvent(url, props)
```

Comment:
`dispatchEvent` will be updated to return the event's payload, which will then be checked here. If it's `mx/load`, then we'll send a post message to the widget here as well.